src/Tickets/Commerce/Order.php

```diff
@@ -514,6 +532,92 @@ public function create( Gateway_Interface $gateway, $args ) {
 	}
 
+	/**
+	 * Filters the values and creates a new Order with Tickets Commerce or updates an existing one.
+	 *
+	 * @since TBD
+	 *
+	 * @param Gateway_Interface $gateway           The gateway to use to create the order.
+	 * @param array             $args              The arguments to create the order.
+	 * @param ?int              $existing_order_id The ID of an existing order to update.
+	 *
+	 * @return false|WP_Post
+	 */
+	public function upsert( Gateway_Interface $gateway, array $args, ?int $existing_order_id = null ) {
+		$gateway_key = $gateway::get_key();
+
+		/**
+		 * Allows filtering of the order upsert arguments for all orders created via Tickets Commerce.
+		 *
+		 * @since TBD
+		 *
+		 * @param array             $args    The arguments to create the order.
+		 * @param Gateway_Interface $gateway The gateway to use to create the order.
+		 */
+		$args = apply_filters( "tec_tickets_commerce_order_{$gateway_key}_upsert_args", $args, $gateway );
+
+		/**
+		 * Allows filtering of the order upsert arguments for all orders created via Tickets Commerce.
+		 *
+		 * @since TBD
+		 *
+		 * @param array             $args    The arguments to create the order.
+		 * @param Gateway_Interface $gateway The gateway to use to create the order.
+		 */
+		$args = apply_filters( 'tec_tickets_commerce_order_upsert_args', $args, $gateway );
+
+		/**
+		 * Allows filtering of the existing order ID before "upserting" an order.
+		 *
+		 * @since TDB
+		 *
+		 * @param int $existing_order_id The existing order ID.
+		 */
+		$existing_order_id = (int) apply_filters( 'tec_tickets_commerce_order_upsert_existing_order_id', $existing_order_id );
+
+		if ( ! $existing_order_id ) {
+			return $this->create( $gateway, $args );
+		}
+
+		/**
+		 * Allows filtering of the order update arguments for all orders created via Tickets Commerce.
+		 *
+		 * @since TBD
+		 *
+		 * @param array             $args
+		 * @param Gateway_Interface $gateway
+		 */
+		$update_args = apply_filters( "tec_tickets_commerce_order_{$gateway_key}_update_args", $args, $gateway );
+
+		/**
+		 * Allows filtering of the order update arguments for all orders created via Tickets Commerce.
+		 *
+		 * @since TBD
+		 *
+		 * @param array             $args
+		 * @param Gateway_Interface $gateway
+		 */
+		$update_args = apply_filters( 'tec_tickets_commerce_order_update_args', $update_args, $gateway );
+
+		$updated = tec_tc_orders()->by_args(
+			[
+				'status' => 'any',
+				'id'     => $existing_order_id,
+			]
+		)->set_args( $update_args )->save();
+
+		if ( empty( $updated[ $existing_order_id ] ) ) {
+			/**
+			 * It seems like the $existing_order_id no longer exists or failed to be updated. Let's create a new one instead.
+			 *
+			 * BE AWARE: THe variable $args here is not passed through the update filters since its going to pass through the create filters.
+			 */
+			return $this->create( $gateway, $args );
+		}
+
+		return tec_tc_get_order( $existing_order_id );
```

Comment:
Make sure the return is what you expect it to be.
It's fine to just forward the return value from it, but take the possible return values into account in the `@return` annotation.

Reply:
`tec_tc_get_order` returns WP_Post|**null** with just the ID as a param - maybe check for empty and return false to ensure the return matches the expected.

Or grab from the repository and return.